transit/transit_types.hpp

```diff
@@ -164,7 +164,7 @@ class Stop
   FeatureIdentifiers m_featureIdentifiers;
   TransferId m_transferId = kInvalidTransferId;
   std::vector<LineId> m_lineIds;
-  m2::PointD m_point;
+  m2::PointD m_point = m2::PointD(0.0, 0.0);
```

Comment:
```suggestion
  m2::PointD m_point {0.0, 0.0};
```

Reply:
Why is this initialization necessary?

Reply:
Inside `Transit_StopSerialization` test, `routing::transit::Stop` object is created, serialized and deserialized. Since this field is uninitialized, some garbage value was written and then read back. Would be unnoticed but `PointUToPointD` does some bounds checking so the deserialized point does not equal to the original.
![image](https://github.com/organicmaps/organicmaps/assets/8740768/b3486705-56d2-4214-85b2-a2f8299fffe3)

Reply:
Well, yes formally, but who cares if the test checks ```TEST(!stop.IsValid(), (stop));```
Some tool shows an error here or some runtime diagnostics?

Reply:
>  Some tool shows an error here or some runtime diagnostics?

The latter. The test also checks `obj.IsEqualForTesting(deserializedObj)` inside `TestSerialization`/`TestCommonSerialization`.
https://github.com/organicmaps/organicmaps/blob/c1de096229597715914cda0f03050a2093416981/transit/transit_tests/transit_test.cpp#L39

Reply:
1. Is the issue limited to tests only?
2. Why 0,0 is a correct initializer? Is a better fix possible?
3. Does it make sense to use some kInvalidPoint check instead of forcing it to zeroes?

Reply:
> 1. Is the issue limited to tests only?

Yeah, for some reason default constructor is used only in tests (`Transit_StopSerialization`).

> 2. Why 0,0 is a correct initializer? Is a better fix possible?
> 3. Does it make sense to use some kInvalidPoint check instead of forcing it to zeroes?

Any value in bounds [-180;180] would work. See `PointDToPointU` for details. Considering that `Transit_StopSerialization` doesn't fail for linux users I assume `m_point={0.0,0.0}` is what's already happening.